pytorch_lightning/loggers/neptune.py

```diff
@@ -296,113 +548,42 @@ def version(self) -> str:
         Returns:
             The id of the experiment if not in offline mode else "offline-id-1234".
         """
-        if self.offline_mode:
-            return "offline-id-1234"
-        return self.experiment.id
+        return self.run._short_id  # skipcq: PYL-W0212
+
+    @staticmethod
+    def _raise_deprecated_api_usage(f_name, sample_code):
+        # TODO: product - review text
+        raise ValueError(f"The function you've used is deprecated.\n"
+                         f"If you are looking for the Neptune logger using legacy Python API it has been renamed to"
+                         f" NeptuneLegacyLogger. The NeptuneLogger was re-written to use the neptune.new Python API"
+                         f" (learn more: https://neptune.ai/blog/neptune-new).\n"
+                         f"Instead of `logger.{f_name}` you can use:\n"
+                         f"\t{sample_code}")
 
     @rank_zero_only
-    def log_metric(
-        self, metric_name: str, metric_value: Union[torch.Tensor, float, str], step: Optional[int] = None
-    ) -> None:
-        """
-        Log metrics (numeric values) in Neptune experiments.
-
-        Args:
-            metric_name: The name of log, i.e. mse, loss, accuracy.
-            metric_value: The value of the log (data-point).
-            step: Step number at which the metrics should be recorded, must be strictly increasing
-        """
-        if is_tensor(metric_value):
-            metric_value = metric_value.cpu().detach()
-
-        if step is None:
-            self.experiment.log_metric(metric_name, metric_value)
-        else:
-            self.experiment.log_metric(metric_name, x=step, y=metric_value)
+    def log_metric(self, *args, **kwargs):
```

Comment:
Why is this depreciated ?

Reply:
This is not a path we want to take for now.

All the loggers are exposing those functions and Netpune Logger shouldn't be different as it will involved learning a new API for Lightning users.

Please, add support back for all the `log_{}` and maybe add a warning that users can use log directly if needed.

Reply:
I don't see any definitions of `log_metric`, `log_text`, `log_image` etc... in any other loggers.

Those methods were available in previous version of logger and we've kept them to inform user that this is deprecated usecase only.

Reply:
It's related to [this comment](https://github.com/PyTorchLightning/pytorch-lightning/pull/6867#discussion_r699224411) I guess.

Reply:
Yes, your integration is definitely in advance compared to other loggers.
This syntax is related to this issue: https://github.com/PyTorchLightning/pytorch-lightning/issues/6720

Reply:
Ok, if you have official plans for that I see no problems with keeping support for those methods.
I'll try to implement them and add some tests.

Reply:
@tchaton 

> Yes, your integration is definitely in advance compared to other loggers.
> But we are in the process to add this `log_{}` for all loggers
> The issue is there: #6720
> Let's make sure that using Netpune `log_{}` works and don't raise an error.

To be sure when you said "works and don't raise an error." do you mean:
a) Remove the implementation (prepared for adding it in the future, existing users will get an error)
b) Keep the method, don't raise an error, print warning informing that if they are looking for the previous implementation they can find here (in essence what is now, but without the first sentence).
c) Keep the method, with empty implementation
d) Something else?

Once the API is there we will be definitely implementing those on our side, but as far as I understand the API is not yet finalized?

Reply:
@tchaton ping on the question here ☝🏻 :)

Reply:
is it possible there was a confusion here between `log_metric` (neptune specific) and `log_metrics` (interface of the base class)?

Reply:
> is it possible there was a confusion here between log_metric (neptune specific) and log_metrics (interface of the base class)?

That was my initial guess, but after @tchaton send us https://github.com/PyTorchLightning/pytorch-lightning/issues/6720  issue I guess that some of the methods like: `log_metric`, `log_text` or `log_image` may be part of the logging API, so we wanted to avoid situation when we:
* first implemented those methods in our old `NeptuneLogger`
* then make them deprecated in our new `NeptuneLogger`
* and then revoke them again as they will be part of new Logger API.

It'd be nice if we implemented them now, but we can't since we don't know list of exptected arguments.

So maybe the best solution is just keeping raise of exception when methods are used, but instead of information that they are deprecated, we could add some more detailed information that they were part of old `NeptuneLogger`, which currently is available in separate repo (according to your request), but we could also mention that those methods **may** be part of some future API.

Eventually we can just drop those functions, so they are not part of new `NeptuneLogger` at all...
This deprecation info is not very useful in fact since someone who imports new `NeptuneLogger` and runs it against old code, will be warned after passing wrong arguments to `__Init__` function.

What do you think?

Reply:
Yes, I believe `log_metrics` is used across Lightning. This should be the API proposed by the NetpuneLogger.

@Herudaio

What I meant is `log_artifact`, `log_image`, etc... were supported in the previous implementation and should still be supported with the new one.

Otherwise, it would be a big BC change for the users.

Best,
T.C

Reply:
> I don't believe a warning would even be required there as they shouldn't be a depreciation.
> The users are relying on the Lightning API of a logger, not the Neptune API.
> 
> Best,
> T.C

Hmm, the initial addition of `log_artifact`, `log_image` on the logger side was the aim of 1:1 parity with Neptune API. As @shnela mentioned at that time there was no Lightning API for that. 

@tchaton could you point us what API (parameters, return type) for those `log_{}` methods should look like? We only found one closed pull request with API proposition for `log_images`, but maybe we are not aware of something. Could you provide a link where we can find what the method signatures should be?

Reply:
I think all we want (if I understand @tchaton correctly) is that you handle either one of these cases:

a) your intention is to remove log_x -> deprecate it in 1.5, remove in version 1.7 but until then provide the functionality if possible
b) your intention is not to remove it -> then keep api the same, no action required

The priority here is to make it as backward compatible as possible. The whole discussion about what log_x might look like in the future for Lightning is out of scope in my opinion can be discussed separately elsewhere.